### PR TITLE
Fix AOT build, replace angular scaffolding in favour of CLI

### PR DESCRIPTION
**Pull request by sandman21dan**
Fixed AOT build

Greatly simplified the app scaffolding by using the angular CLI, which gives us an unified way of building apps and modules.

Used angular-library-builder to accomplish up to date building of shareable modules with complete metadata

### CODE REVIEW

**Comment by sandman21dan** [comment]
You can check the update working by using 

`npm i --save ng2-datepicker-aot@0.0.1`

File: package.json [hunk 1]
```diff
@@ -1,111 +1,54 @@
...
-    "start": "npm run serve",
-    "roll": "node ./scripts/index.js --build",
-    "roll:prerender": "npm run roll && npm run prerender",
-    "prerender": "node ./scripts/index.js --prerender",
-    "clean": "./scripts/clean.sh",
-    "test": "karma start config/karma.conf.js",
-    "lint": "npm run tslint \"src/**/*.ts\"",
-    "tslint": "tslint",
-    "sass": "node-sass src/ng2-datepicker/ng2-datepicker.sass src/ng2-datepicker/ng2-datepicker.css",
-    "build": "npm run css && ./node_modules/.bin/ngc -p tsconfig.build.json && rollup -c",
```
Comment (jkuri) [inline package.json]: can you preserve that command please, and also prepublish.

**Comment by jkuri** [comment]
hi @sandman21dan thanks for your effort on this. I will check this later and try to run locally and if everything works I'll merge this. Can you please bump version in `package.json` so I can publish this later?

File: package.json (hunk 1 above)
Comment (sandman21dan) [inline package.json]: Sorry, which command specifically besides prepublish?

**Comment by sandman21dan** [comment]
@jkuri Bumped version, brough back lots of bits lost from package.json renamed package to original name, and bumped the version.

Should be good to go

Let me know about any issues you might encounter

**Comment by jkuri** [comment]
hi @sandman21dan I am just checking this and when trying to the demo I got `ERROR in AppModule is not an NgModule`, can you fix it please?

**Comment by jkuri** [comment]
sorry, had to reinstall packages - it does work.
one more thing, can you preserve `.sass` styles and include it in `ng2-datepicker.component.ts` as `.sass` file and remove the compiled `.css` one?

**Comment by sandman21dan** [comment]
@jkuri All done, seems like everything works just as expected (neved tried sass rather than scss on the CLI before)

**Comment by jkuri** [comment]
thanks, I would add bundle too but I can add this later. merging.

**Pull request merged**